test/rofl/common/crofsock/crofsocktest.cpp

```diff
@@ -17,13 +17,17 @@ using namespace rofl::openflow;
 
 CPPUNIT_TEST_SUITE_REGISTRATION(crofsocktest);
```

Comment:
asan detects the following two issues

```
$ ./crofsocktest 
.ROUND (0) START
ROUND (0) END
ROUND (1) START
ROUND (1) END
.F


!!!FAILURES!!!
Test Results:
Run:  2   Failures: 1   Errors: 0


1) test: crofsocktest::test_tls (F) line: 141 crofsocktest.cpp
assertion failed
- Expression: timeout > 0



=================================================================
==25302==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 744 byte(s) in 1 object(s) allocated from:
    #0 0x511260 in operator new(unsigned long) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x511260)
    #1 0x5197a0 in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:167:15
    #2 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #3 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #4 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #5 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #6 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

Direct leak of 150 byte(s) in 1 object(s) allocated from:
    #0 0x511260 in operator new(unsigned long) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x511260)
    #1 0x7fa08c89a94c  (/lib64/libstdc++.so.6+0x11794c)
    #2 0x7fa084cf325f  (<unknown module>)

Direct leak of 112 byte(s) in 1 object(s) allocated from:
    #0 0x4d7330 in __interceptor_malloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7330)
    #1 0x7fa08d109b48 in CRYPTO_zalloc (/lib64/libcrypto.so.1.1+0x160b48)

Indirect leak of 65536 byte(s) in 1 object(s) allocated from:
    #0 0x4d7538 in calloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7538)
    #1 0x5338a2 in rofl::cmemory::mallocate(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:232:32
    #2 0x533563 in rofl::cmemory::cmemory(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:24:5
    #3 0x556464 in rofl::crofsock::crofsock(rofl::crofsock_env*) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:47:57
    #4 0x5197d4 in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:167:19
    #5 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #6 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #7 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #8 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #9 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

Indirect leak of 65536 byte(s) in 1 object(s) allocated from:
    #0 0x4d7538 in calloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7538)
    #1 0x5338a2 in rofl::cmemory::mallocate(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:232:32
    #2 0x533563 in rofl::cmemory::cmemory(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:24:5
    #3 0x55613b in rofl::crofsock::crofsock(rofl::crofsock_env*) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:43:35
    #4 0x5197d4 in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:167:19
    #5 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #6 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #7 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #8 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #9 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

Indirect leak of 2600 byte(s) in 21 object(s) allocated from:
    #0 0x4d7330 in __interceptor_malloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7330)
    #1 0x7fa08d109b48 in CRYPTO_zalloc (/lib64/libcrypto.so.1.1+0x160b48)

Indirect leak of 1024 byte(s) in 1 object(s) allocated from:
    #0 0x4d7330 in __interceptor_malloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7330)
    #1 0x7fa08d15def5 in OPENSSL_sk_dup (/lib64/libcrypto.so.1.1+0x1b4ef5)

Indirect leak of 1024 byte(s) in 1 object(s) allocated from:
    #0 0x4d7755 in realloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7755)
    #1 0x7fa08d15dc98 in OPENSSL_sk_insert (/lib64/libcrypto.so.1.1+0x1b4c98)

Indirect leak of 352 byte(s) in 1 object(s) allocated from:
    #0 0x511260 in operator new(unsigned long) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x511260)
    #1 0x5b9a95 in __gnu_cxx::new_allocator<rofl::crofqueue>::allocate(unsigned long, void const*) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/ext/new_allocator.h:111:27
    #2 0x5b9a3b in std::allocator_traits<std::allocator<rofl::crofqueue> >::allocate(std::allocator<rofl::crofqueue>&, unsigned long) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/alloc_traits.h:436:20
    #3 0x5b99e2 in std::_Vector_base<rofl::crofqueue, std::allocator<rofl::crofqueue> >::_M_allocate(unsigned long) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/stl_vector.h:172:20
    #4 0x5b987e in std::_Vector_base<rofl::crofqueue, std::allocator<rofl::crofqueue> >::_M_create_storage(unsigned long) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/stl_vector.h:187:33
    #5 0x5b96b4 in std::_Vector_base<rofl::crofqueue, std::allocator<rofl::crofqueue> >::_Vector_base(unsigned long, std::allocator<rofl::crofqueue> const&) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/stl_vector.h:138:9
    #6 0x591d96 in std::vector<rofl::crofqueue, std::allocator<rofl::crofqueue> >::vector(unsigned long, std::allocator<rofl::crofqueue> const&) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/stl_vector.h:284:9
    #7 0x556369 in rofl::crofsock::crofsock(rofl::crofsock_env*) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:46:37
    #8 0x5197d4 in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:167:19
    #9 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #10 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #11 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #12 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #13 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

Indirect leak of 140 byte(s) in 1 object(s) allocated from:
    #0 0x511260 in operator new(unsigned long) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x511260)
    #1 0x7fa08c8ab254 in void std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<char const*>(char const*, char const*, std::forward_iterator_tag) (/lib64/libstdc++.so.6+0x128254)

Indirect leak of 126 byte(s) in 3 object(s) allocated from:
    #0 0x511260 in operator new(unsigned long) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x511260)
    #1 0x7fa08c8a89fe in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_assign(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (/lib64/libstdc++.so.6+0x1259fe)

Indirect leak of 110 byte(s) in 1 object(s) allocated from:
    #0 0x4d7538 in calloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7538)
    #1 0x5338a2 in rofl::cmemory::mallocate(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:232:32
    #2 0x533563 in rofl::cmemory::cmemory(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:24:5
    #3 0x52b625 in rofl::csockaddr::csockaddr() /home/tobi/workspace/rofl-common/src/rofl/common/csockaddr.h:71:9
    #4 0x555d52 in rofl::crofsock::crofsock(rofl::crofsock_env*) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:30:11
    #5 0x5197d4 in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:167:19
    #6 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #7 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #8 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #9 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #10 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

Indirect leak of 110 byte(s) in 1 object(s) allocated from:
    #0 0x4d7538 in calloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7538)
    #1 0x5338a2 in rofl::cmemory::mallocate(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:232:32
    #2 0x533563 in rofl::cmemory::cmemory(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:24:5
    #3 0x52b625 in rofl::csockaddr::csockaddr() /home/tobi/workspace/rofl-common/src/rofl/common/csockaddr.h:71:9
    #4 0x555d31 in rofl::crofsock::crofsock(rofl::crofsock_env*) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:30:11
    #5 0x5197d4 in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:167:19
    #6 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #7 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #8 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #9 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #10 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

Indirect leak of 110 byte(s) in 1 object(s) allocated from:
    #0 0x4d7538 in calloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7538)
    #1 0x5338a2 in rofl::cmemory::mallocate(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:232:32
    #2 0x533563 in rofl::cmemory::cmemory(unsigned long) /home/tobi/workspace/rofl-common/src/rofl/common/cmemory.cc:24:5
    #3 0x52b625 in rofl::csockaddr::csockaddr() /home/tobi/workspace/rofl-common/src/rofl/common/csockaddr.h:71:9
    #4 0x555d73 in rofl::crofsock::crofsock(rofl::crofsock_env*) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:30:11
    #5 0x5197d4 in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:167:19
    #6 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #7 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #8 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #9 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #10 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

Indirect leak of 40 byte(s) in 1 object(s) allocated from:
    #0 0x4d7330 in __interceptor_malloc (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4d7330)
    #1 0x7fa08d15deb8 in OPENSSL_sk_dup (/lib64/libcrypto.so.1.1+0x1b4eb8)

Indirect leak of 16 byte(s) in 1 object(s) allocated from:
    #0 0x511260 in operator new(unsigned long) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x511260)
    #1 0x549eeb in __gnu_cxx::new_allocator<unsigned int>::allocate(unsigned long, void const*) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/ext/new_allocator.h:111:27
    #2 0x549e8b in std::allocator_traits<std::allocator<unsigned int> >::allocate(std::allocator<unsigned int>&, unsigned long) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/alloc_traits.h:436:20
    #3 0x549a42 in std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_M_allocate(unsigned long) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/stl_vector.h:172:20
    #4 0x5b9f8e in std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_M_create_storage(unsigned long) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/stl_vector.h:187:33
    #5 0x5b9dc4 in std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_Vector_base(unsigned long, std::allocator<unsigned int> const&) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/stl_vector.h:138:9
    #6 0x591e26 in std::vector<unsigned int, std::allocator<unsigned int> >::vector(unsigned long, std::allocator<unsigned int> const&) /usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/bits/stl_vector.h:284:9
    #7 0x5563b8 in rofl::crofsock::crofsock(rofl::crofsock_env*) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:46:58
    #8 0x5197d4 in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:167:19
    #9 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #10 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #11 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #12 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #13 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

SUMMARY: AddressSanitizer: 137730 byte(s) leaked in 38 allocation(s).
```

the other one is:
```
$ ./crofsocktest 
.ROUND (0) START
ROUND (0) END
ROUND (1) START
ROUND (1) END
.=================================================================
==25209==ERROR: AddressSanitizer: heap-use-after-free on address 0x60e000007120 at pc 0x000000453184 bp 0x7fc0ad1f3650 sp 0x7fc0ad1f2e00
READ of size 2 at 0x60e000007120 thread T9
    #0 0x453183 in __interceptor_strlen.part.30 (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x453183)
    #1 0x7fc0b4d494f4 in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&) (/lib64/libstdc++.so.6+0xd54f4)
    #2 0x7fc0b4d2a025 in std::runtime_error::runtime_error(char const*) (/lib64/libstdc++.so.6+0xb6025)
    #3 0x52124c in rofl::exception::exception(rofl::exception const&) /home/tobi/workspace/rofl-common/src/rofl/common/exception.hpp:39:35
    #4 0x593886 in rofl::eLibCall::eLibCall(rofl::eLibCall const&) /home/tobi/workspace/rofl-common/src/rofl/common/exception.hpp:401:7
    #5 0x56b505 in rofl::crofsock::tls_init_context() /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:687:11
    #6 0x567bf3 in rofl::crofsock::tls_connect(bool) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:896:5
    #7 0x57d634 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1721:21
    #8 0x57b964 in rofl::crofsock::handle_write_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1515:5
    #9 0x5cd9f1 in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:700:22
    #10 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #11 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)
    #12 0x7fc0b5b6b36c in start_thread (/lib64/libpthread.so.0+0x736c)
    #13 0x7fc0b4282bbe in __GI___clone (/lib64/libc.so.6+0x110bbe)

0x60e000007120 is located 0 bytes inside of 150-byte region [0x60e000007120,0x60e0000071b6)
freed by thread T7 here:
    #0 0x511f60 in operator delete(void*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x511f60)
    #1 0x522981 in rofl::exception::what() const /home/tobi/workspace/rofl-common/src/rofl/common/exception.hpp:289:5
    #2 0x521240 in rofl::exception::exception(rofl::exception const&) /home/tobi/workspace/rofl-common/src/rofl/common/exception.hpp:39:56
    #3 0x593886 in rofl::eLibCall::eLibCall(rofl::eLibCall const&) /home/tobi/workspace/rofl-common/src/rofl/common/exception.hpp:401:7
    #4 0x56b505 in rofl::crofsock::tls_init_context() /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:687:11
    #5 0x55f318 in rofl::crofsock::tls_accept(int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:772:5
    #6 0x55e546 in rofl::crofsock::tcp_accept(int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:482:15
    #7 0x55f307 in rofl::crofsock::tls_accept(int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:767:15
    #8 0x519afb in crofsocktest::handle_listen(rofl::crofsock&) /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:178:12
    #9 0x57c4a4 in rofl::crofsock::handle_read_event_rxthread(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1682:35
    #10 0x57f568 in rofl::crofsock::handle_read_event(rofl::cthread&, int) /home/tobi/workspace/rofl-common/src/rofl/common/crofsock.cc:1669:5
    #11 0x5cd19d in rofl::cthread::run_loop() /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:697:22
    #12 0x5d0e54 in rofl::cthread::start_loop(void*) /home/tobi/workspace/rofl-common/src/rofl/common/./cthread.hpp:316:41
    #13 0x4e4de2 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x4e4de2)

previously allocated by thread T9 here:
    #0 0x511260 in operator new(unsigned long) (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x511260)
    #1 0x7fc0b4d8b94c  (/lib64/libstdc++.so.6+0x11794c)
    #2 0x7fc0ad1f325f  (<unknown module>)

Thread T9 created by T0 here:
    #0 0x43a820 in pthread_create (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x43a820)
    #1 0x5beaa9 in rofl::cthread::start(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:530:9
    #2 0x5be29a in rofl::cthread::pool_initialize(unsigned int, unsigned int, unsigned int) /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:68:42
    #3 0x515c31 in crofsocktest::setUp() /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:21:3
    #4 0x52f979 in CppUnit::TestCaller<crofsocktest>::setUp() /usr/include/cppunit/TestCaller.h:177:15
    #5 0x7fc0b5947671 in CppUnit::TestCaseMethodFunctor::operator()() const (/lib64/libcppunit-1.13.so.0+0x22671)
    #6 0x60b00000003f  (<unknown module>)

Thread T7 created by T0 here:
    #0 0x43a820 in pthread_create (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x43a820)
    #1 0x5beaa9 in rofl::cthread::start(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:530:9
    #2 0x5be29a in rofl::cthread::pool_initialize(unsigned int, unsigned int, unsigned int) /home/tobi/workspace/rofl-common/src/rofl/common/cthread.cpp:68:42
    #3 0x515c31 in crofsocktest::setUp() /home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest.cpp:21:3
    #4 0x52f979 in CppUnit::TestCaller<crofsocktest>::setUp() /usr/include/cppunit/TestCaller.h:177:15
    #5 0x7fc0b5947671 in CppUnit::TestCaseMethodFunctor::operator()() const (/lib64/libcppunit-1.13.so.0+0x22671)
    #6 0x60b00000003f  (<unknown module>)

SUMMARY: AddressSanitizer: heap-use-after-free (/home/tobi/workspace/rofl-common/test/rofl/common/crofsock/crofsocktest+0x453183) in __interceptor_strlen.part.30
Shadow bytes around the buggy address:
  0x0c1c7fff8dd0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c1c7fff8de0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c1c7fff8df0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c1c7fff8e00: fa fa fa fa fa fa fa fa 00 00 00 00 00 00 00 00
  0x0c1c7fff8e10: 00 00 00 00 00 00 00 00 00 00 00 fa fa fa fa fa
=>0x0c1c7fff8e20: fa fa fa fa[fd]fd fd fd fd fd fd fd fd fd fd fd
  0x0c1c7fff8e30: fd fd fd fd fd fd fd fa fa fa fa fa fa fa fa fa
  0x0c1c7fff8e40: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c1c7fff8e50: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c1c7fff8e60: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c1c7fff8e70: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==25209==ABORTING
```

I will add asan to travis